App.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import {
+  AppRegistry,
+  Button,
+  Text,
+  View
+} from 'react-native';
+import { StackNavigator } from 'react-navigation';
+
+class AllActivitiesScreen extends Component {
+  static navigationOptions = {
+    title: 'All Activities',
+  };
+  render() {
+    const { navigate } = this.props.navigation;
```

Comment:
Nit: You can change this to the following:
```
const { navigation: { navigate } } = this.props;
```

This way in the future, if you access more keys from `this.props`, you can do it all in 1 line. For example:
```
const { navigation: { navigate }, someOtherPropKey, yetAnotherPropKey} = this.props;
```